virtcontainers/qemu.go

```diff
@@ -585,6 +584,85 @@ func (q *qemu) vhostFSSocketPath(id string) (string, error) {
 	return utils.BuildSocketPath(store.RunVMStoragePath, id, vhostFSSocket)
 }
 
+func (q *qemu) setupVirtiofsd(timeout int) (remain int, err error) {
+	sockPath, err := q.vhostFSSocketPath(q.id)
+	if err != nil {
+		return 0, err
+	}
+
+	// The daemon will terminate when the vhost-user socket
+	// connection with QEMU closes.  Therefore we do not keep track
+	// of this child process after returning from this function.
+	sourcePath := filepath.Join(kataHostSharedDir, q.id)
+	args := []string{
```

Comment:
This function does three things:

1. constructs arguments.
1. goroutine and channel logic.
1. adjusts the timeout.

If you broke it into sub-functions for (atleast) (1) and (3), you could add some simple unit tests for those.

Reply:
Sounds good. I'll add some UTs for it.

Reply:
I've split the function and added UTs. PTAL.